### PR TITLE
[DB/SQL]: Fix wrong skilllineability_dbc structure

### PR DESCRIPTION
**Pull request by lineagedr**
## Changes Proposed:
-  Fix wrong skilllineability_dbc structure
-  Add a check for the unused FT_NA_BYTE

You'll also need the new skilllineability_dbc data

[skilllineability_dbc.zip](https://github.com/azerothcore/azerothcore-wotlk/files/7028256/skilllineability_dbc.zip)

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/4312

## Tests Performed:

- Ran the worldserver

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by steenburgh** [comment]
Please add testing instructions with expected results - this looks like a very nuanced issue, so not a lot of people are going to be able to test it without specific instructions

File: src/server/shared/DataStores/DBCDatabaseLoader.cpp [hunk 1]
```diff
@@ -93,6 +93,7 @@ char* DBCDatabaseLoader::Load(uint32& records, char**& indexTable)
                     break;
                 case FT_SORT:
                 case FT_NA:
+                case FT_NA_BYTE:
```
Comment (steenburgh) [inline src/server/shared/DataStores/DBCDatabaseLoader.cpp:96]: It looks like this line is in a generic method that's used for all DBC tables. We should make sure to test that no existing DBC functionality was broken

Comment (lineagedr) [reply src/server/shared/DataStores/DBCDatabaseLoader.cpp:96]: @steenburgh Feel free to test the FT_NA and FT_NA_BYTE are both unused nothing should break dbc functionality.

https://github.com/azerothcore/azerothcore-wotlk/blob/4b7454574412cfe23bb5902b9b3d42d2313b3541/src/common/DataStores/DBCFileLoader.h#L17

**Comment by Yehonal** [comment]
Consider also our guide to export DBC in SQL to test this PR: https://www.azerothcore.org/wiki/how-to-import-dbc-data-in-db

**Comment by kaytotes** [comment]
Tested:

- Applied SQL modification.
- Installed DBC data from the attached zip file.
- Started core.

I also exported the data back out from SQL using these updated definitions and threw them in an MPQ patch and had no issues booting the client, creating chars, spell trainers etc. Before this I could not export the data as we only had 13 of 14 columns. 

I assume this structure is from https://github.com/wowdev/WoWDBDefs/blob/master/definitions/SkillLineAbility.dbd#L139-L151 which definitely seems to be more up to date than the wowdev wiki

👍

File: data/sql/updates/pending_db_world/rev_1629670488191888600.sql [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+ALTER TABLE `skilllineability_dbc`
+    CHANGE `MinSkillLineRank` `ExcludeRace` INT NOT NULL DEFAULT 0,
+    CHANGE `SupercededBySpell` `ExcludeClass` INT NOT NULL DEFAULT 0,
+    CHANGE `AcquireMethod` `MinSkillLineRank` INT NOT NULL DEFAULT 0,
+    CHANGE `TrivialSkillLineRankHigh` `SupercededBySpell` INT NOT NULL DEFAULT 0,
+    CHANGE `TrivialSkillLineRankLow` `AcquireMethod` INT NOT NULL DEFAULT 0, 
+    CHANGE `CharacterPoints_1` `TrivialSkillLineRankHigh` INT NOT NULL DEFAULT 0,
+    CHANGE `CharacterPoints_2` `TrivialSkillLineRankLow` INT NOT NULL DEFAULT 0,
+    CHANGE `TradeSkillCategoryID` `CharacterPoints_1` INT NOT NULL DEFAULT 0,
+    ADD COLUMN `CharacterPoints_2` INT NOT NULL DEFAULT 0 AFTER `CharacterPoints_1`;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1629670488191888600.sql:12]: newline

**Comment by FrancescoBorzi** [comment]
> Tested:
> 
> * Applied SQL modification.
> * Installed DBC data from the attached zip file.
> * Started core.
> 
> I also exported the data back out from SQL using these updated definitions and threw them in an MPQ patch and had no issues booting the client, creating chars, spell trainers etc. Before this I could not export the data as we only had 13 of 14 columns.
> 
> I assume this structure is from https://github.com/wowdev/WoWDBDefs/blob/master/definitions/SkillLineAbility.dbd#L139-L151 which definitely seems to be more up to date than the wowdev wiki
> 
> 👍

@kaytotes have you also had a quick look in game?

**Review by Nefertumm — APPROVED** [review]
Seems fine, add a new line at the end of the sql file

**Comment by temperrr** [comment]
@lineagedr Please look at what the reviewers wrote so it can be merged :)

**Pull request closed without merging**